pennylane_lightning/core/src/simulators/lightning_qubit/StateVectorLQubit.hpp

```diff
@@ -687,5 +688,55 @@ class StateVectorLQubit : public StateVectorBase<PrecisionT, Derived> {
             arr[k] *= inv_norm;
         }
     }
+
+    /**
+     * @brief Prepares a single computational basis state.
+     *
+     * @param index Index of the target element.
+     */
+    void setBasisState(const std::size_t index) {
+        auto length = this->getLength();
+        PL_ABORT_IF(index > length - 1, "Invalid index");
+
+        auto *arr = this->getData();
+        for (size_t k = 0; k < length; k++) {
+            arr[k] = {0.0, 0.0};
+        }
+        arr[index] = {1.0, 0.0};
+    }
+
+    /**
+     * @brief Set values for a batch of elements of the state-vector.
+     *
+     * @param values Values to be set for the target elements.
+     * @param indices Indices of the target elements.
+     */
+    void setStateVector(const std::vector<std::size_t> &indices,
+                        const std::vector<ComplexT> &values) {
+        std::unordered_set<std::size_t> index_set(indices.begin(),
+                                                  indices.end());
+        auto num_indices = index_set.size();
+        PL_ABORT_IF(num_indices != indices.size(), "Indices must be unique");
```

Comment:
Do we care that indices are unique (at this point of the pipeline)? If there are duplicates, the last occurence will have precedence.

Reply:
Your call :) Happy to remove this. I just added extra checks as it was asked for extra checks. Just let me know.

Reply:
I don't see a strong need for it either. Also, agree that an `unordered_set` for an array such as this will be heavy regarding performance and memory.

Reply:
Let's remove it then.

Reply:
Removed here: https://github.com/PennyLaneAI/pennylane-lightning/pull/841/commits/c65256ff1680254547d1ed9e1dbce09e8f0f6ca7